File: backend/src/main/java/com/prontudigital/backend/agendamento/entidades/HorarioTrabalho.java
package com.prontudigital.backend.agendamento.entidades;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;
import java.time.LocalTime;
import java.util.UUID;

/**
 * Janela semanal de atendimento do profissional (RF05).
 *
 * <p>Complementa {@link BloqueioRecorrente} pelo lado oposto: o bloqueio diz
 * quando o profissional <em>nao</em> atende dentro do expediente; aqui fica o
 * expediente em si. Um agendamento precisa caber inteiro em uma destas janelas.
 */
@Entity
@Table(name = "horarios_trabalho")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class HorarioTrabalho {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "uuid", unique = true, insertable = false, updatable = false)
    private UUID uuid;

    @Column(name = "profissional_uuid", nullable = false)
    private UUID profissionalUuid;

    /** ISO-8601: 1=Segunda … 6=Sábado, 7=Domingo */
    @Column(name = "dia_semana", nullable = false)
    private Integer diaSemana;

    @Column(name = "hora_inicio", nullable = false)
    private LocalTime horaInicio;

    @Column(name = "hora_fim", nullable = false)
    private LocalTime horaFim;

    @Column(name = "ativo", nullable = false)
    @Builder.Default
    private Boolean ativo = true;

    @CreationTimestamp
    @Column(name = "criado_em", updatable = false)
    private LocalDateTime criadoEm;

    /** Verdadeiro quando o intervalo cabe inteiro nesta janela. */
    public boolean contem(LocalTime inicio, LocalTime fim) {
        return !inicio.isBefore(horaInicio) && !fim.isAfter(horaFim);
    }
}

File: frontend-web/tipos/horarioTrabalho.ts
/** Janela semanal de atendimento do profissional (RF05). */
export interface HorarioTrabalho {
  id: number
  uuid: string
  profissionalUuid: string
  /** ISO-8601: 1=Segunda … 7=Domingo */
  diaSemana: number
  /** HH:mm */
  horaInicio: string
  horaFim: string
  ativo: boolean
}

export interface HorarioTrabalhoRequisicao {
  profissionalUuid: string
  diaSemana: number
  horaInicio: string
  horaFim: string
}

export const DIAS_SEMANA_TRABALHO: {
  valor: number
  label: string
  abrev: string
}[] = [
  { valor: 1, label: 'Segunda-feira', abrev: 'Seg' },
  { valor: 2, label: 'Terça-feira', abrev: 'Ter' },
  { valor: 3, label: 'Quarta-feira', abrev: 'Qua' },
  { valor: 4, label: 'Quinta-feira', abrev: 'Qui' },
  { valor: 5, label: 'Sexta-feira', abrev: 'Sex' },
  { valor: 6, label: 'Sábado', abrev: 'Sáb' },
  { valor: 7, label: 'Domingo', abrev: 'Dom' },
]

export function nomeDiaSemana(dia: number): string {
  return DIAS_SEMANA_TRABALHO.find(d => d.valor === dia)?.label ?? String(dia)
}
